module/zfs/ddt.c

```diff
@@ -423,8 +423,8 @@ ddt_stat_add(ddt_stat_t *dst, const ddt_stat_t *src, uint64_t neg)
 
 	ASSERT(neg == 0 || neg == -1ULL);	/* add or subtract */
 
-	while (d < d_end)
-		*d++ += (*s++ ^ neg) - neg;
+	for (int i = 0; i < d_end - d; i++)
+		d[i] += (s[i] ^ neg) - neg;
```

Comment:
this is good.  If you wanted to make some more improvements while you're here:

* use `sizeof(dst)/sizeof(uint64_t)` instead of `d_end - d`
* type of `neg` should be `int64_t` (or better yet, take a `boolean_t subtract`
* remove the bit-twiddling with `neg`

Reply:
Compiler will expand `d_end - d` to the constant value during the compilation I believe. Regarding `neg` -- I wanted to keep the original code flow. For sure this can be improved, but that will require some performance testing to be sure there are no regressions.